src/rollCall/index.ts

```diff
@@ -1,18 +1,66 @@
-import { esClient } from "elasticsearch";
+import fetch from 'node-fetch';
+import { IndexReleaseRequest, CreateResolvableIndexRequest, ResolvedIndex, RollcallClient } from './types';
+import urljoin from 'url-join';
 
-const ALIAS = "donors";
+export default (
+  configData: {url: string, entity?: string, type?: string, shardPrefix?: string}
+  ): RollcallClient => {
+  const rootUrl = configData.url;
+  const indexEntity = configData?.entity || "donor";
+  const indexType = configData?.type || "centric";
+  const shardPrefix = configData?.shardPrefix || "pgm";
 
-const getNewIndexName = async (programShortName: string) => {
-  const indexName = `programs-${programShortName}-${Math.random()}`;
-  await esClient.indices.create({
-    index: indexName
-  });
-  return indexName;
-};
+  const createNewResolvableIndex = async (programShortName: string, cloneFromReleasedIndex?: boolean): Promise<ResolvedIndex> => {
+    const url = urljoin(`${rootUrl}`, `/indices/create`);
+    
+    const req: CreateResolvableIndexRequest = {
+      shardPrefix: shardPrefix,
+      shard: await formatProgramShortName(programShortName),
+      entity: indexEntity,
+      type: indexType,
+      cloneFromReleasedIndex: cloneFromReleasedIndex || false
+  }
 
-const release = async (index: string) => Promise.resolve();
+    const newResolvedIndex = await fetch(url, { 
+      method: 'POST',
+      body: JSON.stringify(req), 
+      headers: {  "Content-Type": "application/json" }
+    })
+    .then(res => res.json()) as ResolvedIndex;
 
-export default {
-  getNewIndexName: getNewIndexName,
-  release: release
-};
+    return newResolvedIndex;
+  };
+
+  const release = async (resovledIndex: ResolvedIndex): Promise<boolean> =>  { 
+    const url = urljoin(`${rootUrl}`, `/aliases/release`)
```

Comment:
`urljoin(rootUrl, `/aliases/release`)` is good enough :P

Reply:
I know, but it was already written and it's more work to delete...